client/src/client.rs

```diff
@@ -777,6 +777,47 @@ impl Client {
         )
     }
 
+    /// Create a request with sorting and the given filter.
+    ///
+    /// # Errors
+    /// Fails if sending request fails
+    pub fn request_with_sorting_and_filter<R>(
+        &self,
+        request: R,
+        sorting: Sorting,
+        filter: PredicateBox,
+    ) -> QueryHandlerResult<ClientQueryOutput<R>>
+    where
+        R: Query + Into<QueryBox> + Debug,
+        <R::Output as TryFrom<Value>>::Error: Into<eyre::Error>, // Seems redundant
+    {
+        self.request_with_pagination_and_filter_and_sorting(
+            request,
+            Pagination::default(),
+            sorting,
+            filter,
+        )
+    }
```

Comment:
Do we really need this function?

Reply:
Not really, but it's convenient function. Moreover we have `request_with_pagination_and_filter`.

Reply:
We have few places where we have functions like `do_X_and_Y`, `do_X_and_Z` which rely internally on `do_X_and_Y_and_Z`, maybe we can come up with better approach.

Reply:
Maybe a proc macro? Optional arguments used to be a thing in Rust functional.

Reply:
Maybe a proc macro? Optional arguments used to be a thing in Rust functional.

Reply:
My suggestion is to make a builder (called `QueryRequestBuilder`) and to remove query `request_with_*` methods.

Reply:
Agree with builder approach. 
Other option we might consider: [link](https://www.thecodedmessage.com/posts/default-params/).

Reply:
I've opened [an issue](https://github.com/hyperledger/iroha/issues/3124) and mentioned your suggestion there.